superset-frontend/src/dashboard/components/nativeFilters/FilterBar/index.tsx

```diff
@@ -288,6 +288,10 @@ const FilterBar: React.FC<FiltersBarProps> = ({
     }
   }, [JSON.stringify(filters), JSON.stringify(previousFilters)]);
 
+  useEffect(() => {
+    setDataMaskSelected(() => dataMaskApplied);
+  }, [JSON.stringify(dataMaskApplied), setDataMaskSelected]);
```

Comment:
Can you use `dataMaskAppliedText`? It's already converting the JSON to string.

Reply:
Good point! Done